src/microsoft.sheets/spreadsheet_endpoint.bal

```diff
@@ -416,6 +422,8 @@ public type Table client object {
                 self.properties.tableName = newTableName;
                 return ();
             } else {
+                log:printDebug("Error occurred while renaming the table. HTTP Status Code: " + 
+                httpResponse.statusCode.toJsonString() + ", Reason : " + httpResponse.reasonPhrase);
```

Comment:
Shall we use String templates to concat values? 
+ sign is a bit expensive and runs every time irrespective of debug logs enable or not

Reply:
@daneshk  is it correct to say that this a (widely) known behaviour and that we do not recommend Ballerina programmers to use concat?

Reply:
Even if the debug logs are not enabled, by default the string concatenation using the + operator happens. This is because checking for debug logs enabled/disabled happens after string concatenation using the + operator. However, for expressions which are located within the if checks do not get evaluated unless the if checks conditions are satisfied.

Even if we use string templates they also get converted to string concatenation using + operator. Hence there is no advantage of use of string templates instead of + operator in terms of performance.

If we use an anonymous function as arguments for `log:printDebug()` the expressions within the function declaration evaluation happens after check for debug flag. Hence we follow that approach as a solution here.